### PR TITLE
Feature/nexteuropa 10407

### PR DESCRIPTION
**Pull request by voidtek**

### CODE REVIEW

**Comment by voidtek** [comment]
Update this branch with the branch

File: phpcs-ruleset.xml [hunk 1]
```diff
@@ -81,6 +81,9 @@
         <exclude-pattern>profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.webservice.inc</exclude-pattern>
     </rule>
 
+    <!-- Chosen WSDL callback function FPFISPoetryIntegrationRequest do not follow camel case format. -->
```
Comment (ademarco) [inline phpcs-ruleset.xml]: Here we are excluding the whole PoetryCallback.php file while we should only exclude that particular method signature by using partial ignore as the example below:

```
$xmlPackage = new XMLPackage;
// @codingStandardsIgnoreStart
$xmlPackage['error_code'] = get_default_error_code_value();
// @codingStandardsIgnoreEnd
$xmlPackage->send();
```

See https://pear.php.net/manual/en/package.php.php-codesniffer.advanced-usage.php for more info.

File: profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.webservice.inc [hunk 2]
```diff
@@ -4,6 +4,13 @@
  * Webservice definition and behaviour.
  */
 
+// Setting up templates directory path.
+define('TMGMT_POETRY_XML_TEMPLATES_PATH', drupal_get_path('module', 'tmgmt_poetry_mock') . '/xml_templates');
```
Comment (ademarco) [inline profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.webservice.inc]: `tmgmt_poetry` module does not depend from `tmgmt_poetry_mock` so there is something wrong with this constant definition.

File: profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.theme.inc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+function tmgmt_poetry_theme($existing, $type, $theme, $path) {
+  return [
+    'poetry_callback_answer' => [
+      'variables' => [
+        'demande_id' => NULL,
+        'code' => NULL,
+        'message' => NULL,
+      ],
+      'template' => 'poetry-callback-answer',
+      'path' => TMGMT_POETRY_XML_TEMPLATES_PATH,
```
Comment (ademarco) [inline profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.theme.inc]: Template files are stored in a directory of the `tmgmt_poetry_mock` module, we should move these files around.

File: profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.webservice.inc (hunk 2 above)
Comment (ademarco) [inline profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.webservice.inc]: Actually `xml_templates` does not exists in `tmgmt_poetry_mock` module.

**Comment by netlooker** [comment]
This pull request is having a part of the refactoring which was not approved but could be helpful in future.

Together with [~costvit] we decided to create a new PR to merge outcome from other tickets related to NEPT-102.
The final PR is available here - https://github.com/ec-europa/platform-dev/pull/912

**Pull request closed without merging**